examples/02_remote_agent_server/04_convo_with_api_sandboxed_server.py

```diff
@@ -44,10 +45,43 @@
     logger.error("RUNTIME_API_KEY required")
     exit(1)
 
+
+def get_latest_commit_sha(
+    repo: str = "OpenHands/software-agent-sdk", branch: str = "main"
+) -> str:
+    """
+    Return the full SHA of the latest commit on `branch` for the given GitHub repo.
+    Respects an optional GITHUB_TOKEN to avoid rate limits.
+    """
+    url = f"https://api.github.com/repos/{repo}/commits/{branch}"
+    headers = {}
+    token = os.getenv("GITHUB_TOKEN") or os.getenv("GH_TOKEN")
+    if token:
+        headers["Authorization"] = f"Bearer {token}"
+
+    resp = requests.get(url, headers=headers, timeout=20)
+    if resp.status_code != 200:
+        raise RuntimeError(f"GitHub API error {resp.status_code}: {resp.text}")
+    data = resp.json()
+    sha = data.get("sha")
+    if not sha:
+        raise RuntimeError("Could not find commit SHA in GitHub response")
+    logger.info(f"Latest commit on {repo} branch={branch} is {sha}")
+    return sha
+
+
+# If GITHUB_SHA is set (e.g. running in CI of a PR), use that to ensure consistency
+# Otherwise, get the latest commit SHA from main branch (images are built on main)
+server_image_sha = os.getenv("GITHUB_SHA") or get_latest_commit_sha(
+    "OpenHands/software-agent-sdk", "main"
+)
+server_image = f"ghcr.io/openhands/agent-server:{server_image_sha[:7]}-python-amd64"
+logger.info(f"Using server image: {server_image}")
```

Comment:
I don't know if this is necessarily blocking, but this is a bit complex for an example. Maybe we could move this off into some other utility function in another file?